src/huggingface_hub/hf_api.py

```diff
@@ -1726,17 +1737,78 @@ def move_repo(
             " completed."
         )
 
+    def create_commit(
+        self,
+        *,
+        repo_id: str,
+        operations: Iterable[CommitOperation],
+        commit_summary: str,
+        commit_description: Optional[str] = None,
+        token: Optional[str] = None,
+        repo_type: Optional[str] = None,
+        revision: Optional[str] = None,
+        create_pr: Optional[bool] = None,
+    ) -> Optional[str]:
+        commit_description = (
+            commit_description if commit_description is not None else ""
+        )
+        repo_type = repo_type if repo_type is not None else REPO_TYPE_MODEL
+        if repo_type not in REPO_TYPES:
+            raise ValueError(f"Invalid repo type, must be one of {REPO_TYPES}")
+        token, name = self._validate_or_retrieve_token(token)
+        revision = revision if revision is not None else DEFAULT_REVISION
+        create_pr = create_pr if create_pr is not None else False
+
+        additions = [op for op in operations if isinstance(op, CommitOperationAdd)]
+        deletions = [op for op in operations if isinstance(op, CommitOperationDelete)]
+
+        annotated_additions = prepare_file_upload(
+            files=additions,
+            repo_type=repo_type,
+            repo_id=repo_id,
+            token=token,
+            revision=revision,
+            endpoint=self.endpoint,
+        )
+        upload_lfs_files(
+            files=annotated_additions,
+            repo_type=repo_type,
+            repo_id=repo_id,
+            token=token,
+            revision=revision,
+            endpoint=self.endpoint,
+        )
+        commit_payload = prepare_commit_payload(
+            additions=annotated_additions,
+            deletions=deletions,
+            commit_summary=commit_summary,
+            commit_description=commit_description,
+        )
+        commit_url = f"{self.endpoint}/api/{repo_type}s/{repo_id}/commit/{revision}"
+
+        commit_resp = requests.post(
+            url=commit_url,
+            headers={"Authorization": f"Bearer {token}"},
+            json=commit_payload,
+            params={"create_pr": 1} if create_pr else None,
+        )
+        commit_resp.raise_for_status()
+        return commit_resp.json().get("pullRequestUrl", None)
+
     @_deprecate_positional_args
     def upload_file(
         self,
         *,
-        path_or_fileobj: Union[str, bytes, IO],
+        path_or_fileobj: Union[str, bytes, BufferedIOBase],
         path_in_repo: str,
         repo_id: str,
         token: Optional[str] = None,
         repo_type: Optional[str] = None,
         revision: Optional[str] = None,
         identical_ok: bool = True,
+        commit_summary: Optional[str] = None,
```

Comment:
Could we also include these args in the `metadata_update()` function as well, e.g. how I did it here: https://github.com/huggingface/huggingface_hub/pull/884/files#diff-52ca35d2851516051ee1658be7c52fedba606b08b94bdc3ac2bb67f54e0d0a0bR198-R200